util/patricia_trie/src/node.rs

```diff
@@ -107,13 +108,43 @@ impl<'a> Node<'a> {
 	pub fn try_decode_hash(node_data: &[u8]) -> Option<H256> {
 		let r = Rlp::new(node_data);
 		if r.is_data() && r.size() == 32 {
-			Some(r.as_val().expect("Hash is the correct size of 32 bytes; qed"))
+			Some(
+				r.as_val()
+					.expect("Hash is the correct size of 32 bytes; qed"),
+			)
 		} else {
 			None
 		}
 	}
 }
 
+impl<'a> fmt::Debug for Node<'a> {
+	/// Recursion helper for implementation of formatting trait.
+	fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+		match &self {
+			Node::Leaf(slice, value) => f.debug_struct("Node::Leaf")
+				.field("slice", &slice)
+				.field("value", &value.pretty())
+				.finish(),
+			Node::Extension(ref slice, ref item) => {
+				let item = item.fmt(f)?;
+				f.debug_struct("Node::Extension")
+					.field("slice", &slice)
+					.field("item", &item)
+					.finish()
+			}
+			Node::Branch(ref nodes, ref value) => {
+				let nodes_fmt = f.debug_list().entries(nodes).finish()?;
+				f.debug_struct("Node::Branch")
+					.field("nodes", &nodes_fmt)
```

Comment:
I think this is invalid, cause `nodes_fmt` is `()`.? [docs](https://doc.rust-lang.org/std/fmt/struct.DebugList.html#method.finish)